File: backend/app/models/purchase_order.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Float, Integer, DateTime, Text, ForeignKey
from sqlalchemy.orm import relationship
from ..database import Base


class PurchaseOrder(Base):
    __tablename__ = "purchase_orders"

    id = Column(String, primary_key=True, default=lambda: f"OC-{uuid.uuid4().hex[:6].upper()}")
    supplier = Column(String(255), nullable=False)
    item = Column(String(255), nullable=False)
    qty = Column(Integer, nullable=False)
    currency = Column(String(10), default="DOP")
    total = Column(Float, nullable=False)
    status = Column(String(20), default="solicitud", index=True)
    notes = Column(Text, default="")
    created_by = Column(String, ForeignKey("users.id"), nullable=True)
    received_at = Column(DateTime, nullable=True)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))

    creator = relationship("User", back_populates="purchase_orders")
    approvals = relationship("Approval", back_populates="purchase_order", cascade="all, delete-orphan")

File: dotnet-backend/src/Javaline.Commercial.Domain/Entities/PurchaseOrder.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using Javaline.Commercial.Domain.Common;

namespace Javaline.Commercial.Domain.Entities;

[Table("purchase_orders")]
public class PurchaseOrder : BaseEntity
{
    [Key]
    [Column("id")]
    public new string Id { get; set; } = Guid.NewGuid().ToString();

    [Column("supplier")]
    public string Supplier { get; set; } = string.Empty;

    [Column("item")]
    public string Item { get; set; } = string.Empty;

    [Column("qty")]
    public decimal Qty { get; set; }

    [Column("currency")]
    public string Currency { get; set; } = "DOP";

    [Column("total")]
    public decimal Total { get; set; }

    [Column("status")]
    public string Status { get; set; } = "pending";

    [Column("notes")]
    public string? Notes { get; set; }

    [Column("created_by")]
    public string? CreatedBy { get; set; }

    [Column("received_at")]
    public DateTime? ReceivedAt { get; set; }

    public User? Creator { get; set; }
    [JsonIgnore] public ICollection<Approval> Approvals { get; set; } = new List<Approval>();
}
